sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -965,18 +1000,21 @@ class TriggerDriver(with_metaclass(ABCMeta, object)):
   """Breaks a series of bundle and timer firings into window (pane)s."""
 
   @abstractmethod
-  def process_elements(self, state, windowed_values, output_watermark):
+  def process_elements(self, state, windowed_values, output_watermark,
+                       input_watermark=MIN_TIMESTAMP):
     pass
 
   @abstractmethod
-  def process_timer(self, window_id, name, time_domain, timestamp, state):
+  def process_timer(self, window_id, name, time_domain, timestamp, state,
+                    input_watermark=None):
     pass
 
   def process_entire_key(
-      self, key, windowed_values, output_watermark=MIN_TIMESTAMP):
+      self, key, windowed_values, input_watermark=MIN_TIMESTAMP,
```

Comment:
Have we verified we found all callers of this code? Inserting a (default) argument before existing arguments could result in an off-by-one error for unmodified callers. 

Also, the ordering should be consistent with `process_elements()`.

Reply:
Is `MIN_TIMESTAMP` the correct default?

Reply:
Yeah, the only other place where these are called directly are in the Batch Dataflow Python Worker. So by setting these to MIN_TIMESTAMP we go back to the old implementation which didn't affect Batch.